devtools/gossipwith.c

```diff
@@ -137,11 +156,21 @@ static struct io_plan *handshake_success(struct io_conn *conn,
 				OPTIONAL_FEATURE(OPT_INITIAL_ROUTING_SYNC));
 
 	if (!no_init) {
-		msg = towire_init(NULL, NULL, features, NULL);
+		struct tlv_init_tlvs *tlvs = NULL;
+		if (blockchainparams) {
+			tlvs = tlv_init_tlvs_new(NULL);
+			tlvs->networks = tal(tlvs, struct tlv_init_tlvs_networks);
+			tlvs->networks->chains = tal_arr(tlvs->networks, struct bitcoin_blkid, 1);
+			tlvs->networks->chains[0] = blockchainparams->genesis_blockhash;
+			msg = towire_init(NULL, NULL, features, tlvs);
+		} else
+			msg = towire_init(NULL, NULL, features, NULL);
```

Comment:
These lines can be pulled out of the `if`-statement and combined.

Reply:
How could we `blockchainparams->genesis_blockhash;` if `blockchainparams` isn't defined ?

Reply:
No I mean we can make this:

```c
			msg = towire_init(NULL, NULL, features, tlvs);
		} else
			msg = towire_init(NULL, NULL, features, NULL);
```
into this
```c
		}
		msg = towire_init(NULL, NULL, features, tlvs);
```

Since `tlvs` is initialized to `NULL` anyway.

Reply:
Ah, yeah absolutely, I..